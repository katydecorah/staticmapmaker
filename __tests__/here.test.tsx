import { render } from "@testing-library/react";
import Here from "@/pages/here";

it("renders homepage unchanged", () => {
  const { container } = render(<Here />);
  expect(container).toMatchSnapshot();
});
