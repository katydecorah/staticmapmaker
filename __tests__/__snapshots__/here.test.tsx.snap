// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="col-left"
    >
      <div
        class="sticky"
      >
        <header
          class="header"
        >
          <div
            class="title"
          >
            <h1>
              <a
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="header-icon"
                  data-icon="map"
                  focusable="false"
                  height="128"
                  role="img"
                  viewBox="0 0 128 128"
                  width="128"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 0C2.367 0 0 2.367 0 5.5v117c0 3.133 2.367 5.5 5.5 5.5h117c3.133 0 5.5-2.367 5.5-5.5V5.5c0-3.133-2.367-5.5-5.5-5.5H5.5zM8 8h112v112H8V8zm35.188 2l15 15-9.594 9.594a2 2 0 1 0 2.813 2.813L61 27.813 101.19 68l-50.032 50h5.688l47.156-47.187 14 14v-5.625L106.814 68l11.188-11.188v-5.625l-14 14L63.814 25l15-15H73.19L61 22.186 48.814 10h-5.625zM48 52c-8.8 0-16 7.2-16 16 0 12 16 32 16 32s16-20 16-32c0-8.8-7.2-16-16-16zm0 8c4.4 0 8 3.6 8 8s-3.6 8-8 8-8-3.6-8-8 3.6-8 8-8z"
                  />
                </svg>
                <div>
                  HERE
                   Static Map Maker
                </div>
              </a>
            </h1>
          </div>
        </header>
        <div
          class="controls"
        >
          <div
            class="container"
          >
            <div
              class="error fieldset-row"
            >
              <label
                class="label"
                for="api"
              >
                API Key
              </label>
              <input
                id="api"
                placeholder="API Key (required)"
                type="text"
                value=""
              />
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="location"
              >
                Location
              </label>
              <input
                id="location"
                type="text"
                value="42.6564,-73.7638"
              />
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="zoom"
              >
                Zoom
              </label>
              <input
                id="zoom"
                max="22"
                min="0"
                type="range"
                value="13"
              />
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="width"
              >
                Width
              </label>
              <input
                id="width"
                max="900"
                min="80"
                type="number"
                value="600"
              />
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="height"
              >
                Height
              </label>
              <input
                id="height"
                max="834"
                min="80"
                type="number"
                value="300"
              />
            </div>
            <div
              class="fieldset-side fieldsetRow"
            >
              <div
                class="checkbox-fieldset"
              >
                <input
                  id="dot"
                  type="checkbox"
                />
                <label
                  class="checkbox-label"
                  for="dot"
                >
                  No dot
                </label>
              </div>
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="style"
              >
                Map style
              </label>
              <select
                id="style"
              >
                <option
                  value="alps"
                >
                  alps
                </option>
                <option
                  value="dreamworks"
                >
                  dreamworks
                </option>
                <option
                  value="flame"
                >
                  flame
                </option>
                <option
                  value="fleet"
                >
                  fleet
                </option>
                <option
                  value="mini"
                >
                  mini
                </option>
              </select>
            </div>
            <div
              class="fieldset-side fieldsetRow"
            >
              <div
                class="checkbox-fieldset"
              >
                <input
                  id="map-header"
                  type="checkbox"
                />
                <label
                  class="checkbox-label"
                  for="map-header"
                >
                  Map header
                </label>
              </div>
            </div>
            <div
              class="fieldset-side fieldsetRow"
            >
              <div
                class="checkbox-fieldset"
              >
                <input
                  id="no-crop"
                  type="checkbox"
                />
                <label
                  class="checkbox-label"
                  for="no-crop"
                >
                  Don't crop labels
                </label>
              </div>
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="rotate"
              >
                Rotate
              </label>
              <input
                id="rotate"
                max="360"
                min="0"
                type="range"
                value="0"
              />
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="format"
              >
                Format
              </label>
              <select
                id="format"
              >
                <option
                  value="0"
                >
                  PNG
                </option>
                <option
                  value="1"
                >
                  JPEG
                </option>
                <option
                  value="2"
                >
                  GIF
                </option>
                <option
                  value="3"
                >
                  BMP
                </option>
                <option
                  value="4"
                >
                  PNG8
                </option>
                <option
                  value="5"
                >
                  SVG (only for companylogo)
                </option>
              </select>
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="ppi"
              >
                Resolution
              </label>
              <select
                id="ppi"
              >
                <option
                  value="72"
                >
                  72 (default)
                </option>
                <option
                  value="250"
                >
                  250 (mobile)
                </option>
                <option
                  value="320"
                >
                  320 (hi-res)
                </option>
                <option
                  value="500"
                >
                  500
                </option>
              </select>
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="scalebar"
              >
                Scalebar
              </label>
              <select
                id="scalebar"
              >
                <option
                  value=""
                />
                <option
                  value="k"
                >
                  kilometers/meters
                </option>
                <option
                  value="m"
                >
                  miles/feet
                </option>
                <option
                  value="km"
                >
                  both
                </option>
              </select>
            </div>
            <div
              class="fieldset-side fieldsetRow"
            >
              <div
                class="checkbox-fieldset"
              >
                <input
                  id="pip"
                  type="checkbox"
                />
                <label
                  class="checkbox-label"
                  for="pip"
                >
                  Enable picture in picture
                </label>
              </div>
            </div>
            <div
              class="fieldset-row"
            >
              <label
                class="label"
                for="language"
              >
                Language
              </label>
              <select
                id="language"
              >
                <option
                  value=""
                />
                <option
                  value="ara"
                >
                  Arabic
                </option>
                <option
                  value="chi"
                >
                  Chinese (simplified)
                </option>
                <option
                  value="cht"
                >
                  Chinese (traditional)
                </option>
                <option
                  value="dut"
                >
                  Dutch
                </option>
                <option
                  value="eng"
                >
                  English
                </option>
                <option
                  value="fre"
                >
                  French
                </option>
                <option
                  value="ger"
                >
                  German
                </option>
                <option
                  value="gle"
                >
                  Gaelic
                </option>
                <option
                  value="gre"
                >
                  Greek
                </option>
                <option
                  value="heb"
                >
                  Hebrew
                </option>
                <option
                  value="hin"
                >
                  Hindi
                </option>
                <option
                  value="ind"
                >
                  Indonesian
                </option>
                <option
                  value="ita"
                >
                  Italian
                </option>
                <option
                  value="per"
                >
                  Persian
                </option>
                <option
                  value="pol"
                >
                  Polish
                </option>
                <option
                  value="por"
                >
                  Portuguese
                </option>
                <option
                  value="rus"
                >
                  Russian
                </option>
                <option
                  value="sin"
                >
                  Sinhalese
                </option>
                <option
                  value="spa"
                >
                  Spanish
                </option>
                <option
                  value="tha"
                >
                  Thai
                </option>
                <option
                  value="tur"
                >
                  Turkish
                </option>
                <option
                  value="ukr"
                >
                  Ukranian
                </option>
                <option
                  value="urd"
                >
                  Urdu
                </option>
                <option
                  value="vie"
                >
                  Vietnamese
                </option>
                <option
                  value="wel"
                >
                  Welsh
                </option>
              </select>
            </div>
            <div
              class="fieldset-side"
            >
              <button
                class="btn"
              >
                <svg
                  aria-hidden="true"
                  class="feather"
                  fill="none"
                  focusable="false"
                  height="24"
                  role="img"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                 Add a 
                POI
              </button>
            </div>
          </div>
        </div>
        <div
          class="try-another"
        >
          <h2>
            Try another API
          </h2>
          <div
            class="try-another-btn-group"
          >
            <a
              class="btn"
              href="/bing"
            >
              <svg
                aria-hidden="true"
                class="feather"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
               
              Bing
            </a>
            <a
              class="btn"
              href="/google"
            >
              <svg
                aria-hidden="true"
                class="feather"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
               
              Google
            </a>
            <a
              class="btn"
              href="/google-street-view"
            >
              <svg
                aria-hidden="true"
                class="feather"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
               
              Google Street View
            </a>
            <a
              class="btn"
              href="/mapbox"
            >
              <svg
                aria-hidden="true"
                class="feather"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
               
              Mapbox
            </a>
            <a
              class="btn"
              href="/mapquest"
            >
              <svg
                aria-hidden="true"
                class="feather"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
               
              MapQuest
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-right"
    >
      <div
        class="static"
      >
        <div
          class="static-wrapper api-required"
        >
          <img
            alt="HERE Map of 42.6564,-73.7638"
            class="static-map"
            height="300"
            id="map"
            src="/img/here-placeholder.png"
            width="600"
          />
        </div>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="info"
        >
          <h2>
            Request
          </h2>
          <div
            class="code"
          >
            <pre>
              https://image.maps.ls.hereapi.com/mia/1.6/mapview?c=42.6564%2C-73.7638&w=600&h=300&z=13&nodot=false&style=flame&f=1&ppi=72&apiKey=YOUR-API-TOKEN-HERE
            </pre>
          </div>
          Display the map as an image:
          <div
            class="code"
          >
            <pre>
              &lt;img width="
              600
              " src="
              https://image.maps.ls.hereapi.com/mia/1.6/mapview?c=42.6564%2C-73.7638&w=600&h=300&z=13&nodot=false&style=flame&f=1&ppi=72&apiKey=YOUR-API-TOKEN-HERE
              " alt="
              HERE map of 42.6564,-73.7638
              " /&gt;
            </pre>
          </div>
          Display the map as a CSS background image:
          <div
            class="code"
          >
            <pre>
              background-image: url(
              https://image.maps.ls.hereapi.com/mia/1.6/mapview?c=42.6564%2C-73.7638&w=600&h=300&z=13&nodot=false&style=flame&f=1&ppi=72&apiKey=YOUR-API-TOKEN-HERE
              );
            </pre>
          </div>
          <h2>
            How to use
          </h2>
          <ol>
            <li>
              Get a 
              <a
                href="https://developer.here.com/documentation/map-image/dev_guide/topics/request-format.html"
              >
                HERE
                 API key
              </a>
              .
            </li>
            <li>
              Customize the fields to build your map.
            </li>
            <li>
              Copy and paste the generated code into your HTML document or a stylesheet.
            </li>
          </ol>
          <h2>
            About
          </h2>
          <p>
            You can add a static map to any website with simple HTML and no JavaScript. Using URL parameters, you can describe the type of map that you want and 
            HERE
             will send it back.
          </p>
          <p>
            The Static Map Maker helps you configure the 
            HERE
             API parameters.
          </p>
          <p>
            Check out 
            <a
              href="https://developer.here.com/documentation/map-image/dev_guide/topics/introduction.html"
            >
              HERE
              's static maps API
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </main>
</div>
`;
